client/protocols/analytics.py

```diff
@@ -8,9 +10,25 @@
 # TODO(albert): rename this InformationProtocol
 # Add all command line arguments here
 
+log = logging.getLogger(__name__)
+
 class AnalyticsProtocol(models.Protocol):
     """A Protocol that analyzes how much students are using the autograder."""
 
+    RE_SNIPPET = re.compile(r"""
+        \s*[\#\;]\s+BEGIN\s+(.*?)\n # \1 is question name
+        (.*)                        # \2 is the contents in between
+        \s*[\#\;]\s+END\s+\1\n
+        """, re.X | re.I | re.S)
+
+    RE_DEFAULT_CODE = re.compile(r"""
+    ^\"\*\*\*\sREPLACE\sTHIS\sLINE\s\*\*\*\"$
+    """, re.X | re.I)
+
+    RE_REPLACE_MARK = re.compile(r"""
+            [\#\;][ ]Replace[ ].+$
```

Comment:
The `[ ].+$` isn't necessary here. I think just checking a prefix `Replace` is fine.